src/System.IO.FileSystem/src/System/IO/FileSystem.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.IO;
+using System.Text;
+
+#if MS_IO_REDIST
+namespace Microsoft.IO
+#else
+namespace System.IO
+#endif
+{
+    internal static partial class FileSystem
+    {
+        internal static string GetDirectoryRoot(string path)
```

Comment:
@JeremyKuhne this internal method was originally inside the Directory class, but it was only used by FileSystem, once in Windows, once in Unix. I was having issues splitting Directory and only importing the partial file with this method, because FileSystem.AccessControl was already consuming the assembly where Directory was published, but I was trying to force compiling an extra internal section of that same class. I fixed it by moving this method to FileSystem, in a new file shared by all architectures.

Reply:
That's fine.